### PR TITLE
warn if some NULLs are not in default

### PR DESCRIPTION
**Pull request by AshesITR**

### CODE REVIEW

File: R/with.R [hunk 1]
```diff
@@ -57,6 +57,16 @@ with_defaults <- function(..., default = default_linters) {
     )
   }
 
+  to_null <- vapply(vals, is.null, logical(1L))
+  if (!all(nms[to_null] %in% names(default))) {
+    bad_nms <- setdiff(nms[to_null], names(default))
+    is_are <- if (length(bad_nms) > 1L) "are" else "is"
+    warning(
+      "Trying to remove ", glue::glue_collapse(sQuote(bad_nms), sep = ", ", last = " and "),
+      ", which ", is_are, " not in `default`."
```
Comment (MichaelChirico) [inline R/with.R:66]: does ?with_defaults describe `default`? otherwise this message could be confusing

**Review by MichaelChirico — APPROVED** [review]
Approving in case there's nothing to add to the warning to help orient the user what 'default' is

File: R/with.R (hunk 1 above)
Comment (AshesITR) [reply R/with.R:66]: It has to for R CMD check to succeed. Here's the roxy:

`#' @param default list of elements to modify.`

Comment (MichaelChirico) [reply R/with.R:66]: 👍 silly misunderstanding on my part, thanks

**Pull request merged**